### DISCUSSION TITLE
Linux (Ubuntu 20.04) Support?

### DISCUSSION THREAD

**Discussion by JoshTheDeveloperr** [joemasilotti/railsdevs.com#761] [Q&A]
Does this repo officially support Linux?

I've spent probably 6 hours and counting trying to fix various errors I'm encountering when running `bin/setup` however I'm having no luck.

All of the non-Ruby packages were installed with no problem with the exception of `stripe/stripe-cli/stripe`. I had to follow this issue thread for help https://github.com/stripe/stripe-cli/issues/645.

This package fixed the Stripe issue `brew install bfontaine/utils/stripe`

_**Disclaimer:** I have absolutely no experience with Ruby._

I currently encountering a ton of errors after running `bin/setup`.

Paste of my errors: https://pastebin.com/TLH6N05N

The errors generally look like:
![image](https://user-images.githubusercontent.com/36740339/215189073-6b6edd18-0f0a-4fe9-92f5-db2ce436c7cf.png)
![image](https://user-images.githubusercontent.com/36740339/215189125-cac4f494-f711-42c8-90e9-efcb5db5a1a8.png)
![image](https://user-images.githubusercontent.com/36740339/215189163-951bd33b-dfc6-40fa-8d43-c4205211ab93.png)

**Comment by joemasilotti** [comment]
I'm sorry to hear you're running into issues setting up the code on your machine!

To be honest, I've never run the codebase on a Linux machine. But based on [this doc change](https://github.com/joemasilotti/railsdevs.com/commit/f70d68f928d82802611407de279a65e47677cf9d) I'm fairly certain at least one other person has.

From the logs, it looks like `http-parser`, `pg`, and `nokogiri` are failing to install because their native extensions can't be loaded.

For `pg`, try following the instructions in the README for Debian. Or something similar for your specific Linux build.

`nokogiri` now includes bundled native extensions. Can you check to see if [your machine is supported](https://nokogiri.org)?

I'm assuming that `http-parser` depends on `nokogiri`. So maybe getting that installed first will help?

**Reply by JoshTheDeveloperr** [reply]
hmm. What host can I use that will deploy this codebase on a macos machine? I'm currently on DigitalOcean.

**Reply by joemasilotti** [reply]
I'm hosting the app on Heroku which I don't think is a macOS machine. But to be honest, I have no idea. That's why I leave it up to Heroku!

**Comment by JoshTheDeveloperr** [comment]
@axhgl How did you manage to deploy this codebase on Linux? And which Linux distro did you use?

**Comment by JoshTheDeveloperr** [comment]
Ok so, I tried reinstalling everything again but this time on Ubuntu 18.04.

**1)** I got this error: `An error occurred while installing pg (1.2.3), and Bundler cannot continue.`. But this was resolved by installing `sudo apt install postgresql-contrib libpq-dev`

**2)**  nokogiri didn't fail to install this time for some reason. I suppose that's because I'm using Ubuntu 18.04? 

**3)** I'm now just left with figuring out this error `An error occurred while installing http-parser (1.2.3), and Bundler cannot continue.`

**Reply by JoshTheDeveloperr** [reply]
I don't understand what's wrong with this error though: **gem install http-parser**
![image](https://user-images.githubusercontent.com/36740339/215286891-91af2b63-f5a0-45ff-8d74-f199a120176d.png)


ffi & ffi-compiler are both installed..

**Reply by joemasilotti** [reply]
It looks like it can't find the right version of ffi, 3.1. Maybe there are some bundler flags you can pass to `bundle install` to point to the right location on your machine?
